crates/rspack_core/src/normal_module_factory.rs

```diff
@@ -633,33 +632,35 @@ impl NormalModuleFactory {
 
     self.context.module_type = Some(resolved_module_type);
 
-    let normal_module = NormalModule::new(
-      request,
-      user_request,
-      dependency.request().to_owned(),
-      resolved_module_type,
-      resolved_parser_and_generator,
-      resolved_parser_options,
-      resolved_generator_options,
-      match_resource_data,
-      resource_data,
-      resolved_resolve_options,
-      loaders,
-      self.context.options.clone(),
-      contains_inline,
-    );
+    let id = NormalModule::create_id(resolved_module_type, &user_request);
 
     let module = if let Some(module) = self
       .plugin_driver
       .module(ModuleArgs {
         dependency_type: data.dependency.dependency_type().clone(),
-        indentfiler: normal_module.identifier(),
+        indentfiler: ModuleIdentifier::from(id.as_str()),
```

Comment:
```suggestion
        created_module: normal_module,
```

align with webpack https://github.com/webpack/webpack/blob/87660921808566ef3b8796f8df61bd79fc026108/lib/NormalModuleFactory.js#L329